account_bank_match/models/account_bank_statement.py

```diff
@@ -254,6 +254,30 @@ def _extract_references(self):
                 self._handle_error(msg)
         return matches
 
+    def get_proper_channel(self):
```

Comment:
Where is this function called??? Anyway, never base functionality on names. And certainly not on hardcoded names like here.
I understand the problem at hand is to match bank statement lines to orders. The orders should have the same company as the bank statement line. Seems logical. I do not see this in this implementation at all. What is your reasoning to solve it this way?

Reply:
When the rules are parsed if the rule cannot be resolved the code searches for a attribute of the class. The relevant rules have been changed to include `('sales_channel_id', 'in', '@get_proper_channel()@')` and that's how this is called

See https://github.com/1200wd/1200wd_addons/blob/5f09ac453ae764fd7edafbc507914cfff39b3d83/account_bank_match/models/account_bank_statement.py#L328
and:
https://github.com/1200wd/1200wd_addons/blob/5f09ac453ae764fd7edafbc507914cfff39b3d83/account_bank_match/models/account_bank_statement.py#L331

> The orders should have the same company as the bank statement line. Seems logical. I do not see this in this implementation at all. What is your reasoning to solve it this way?


The companies will be removed in a later point and all the journals/statements will be one one company, so I cannot use `company_id` to do this.

My reasoning is that I need to produce a solution as less intrusive as possible without changing the complicated code of the parsing of the rules but rather adding an extra rule that will be parsed by my code. I saw this as the most clean way to implement the functionality requested.